File: windows/ReactNativePSPDFKit/ReactNativePSPDFKit/Events/PdfViewDocumentSavedEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json.Linq;
using ReactNative.UIManager.Events;

namespace ReactNativePSPDFKit.Events
{
    /**
     * Event sent by the PDFViewPage when the document save succeeded
     */
    class PdfViewDocumentSavedEvent : Event
    {

        public const string EVENT_NAME = "pdfViewDocumentSaved";

        public PdfViewDocumentSavedEvent(int viewId) : base(viewId)
        {
        }


        public override string EventName => EVENT_NAME;

        public override void Dispatch(RCTEventEmitter rctEventEmitter)
        {
            rctEventEmitter.receiveEvent(ViewTag, EventName, null);
        }
    }
}

File: android/src/main/java/com/pspdfkit/react/events/PdfViewDocumentSavedEvent.java
package com.pspdfkit.react.events;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.uimanager.events.Event;
import com.facebook.react.uimanager.events.RCTEventEmitter;

import androidx.annotation.IdRes;

/**
 * Event sent by the {@link com.pspdfkit.views.PdfView} when the document was saved.
 */
public class PdfViewDocumentSavedEvent extends Event<PdfViewDocumentSavedEvent> {

    public static final String EVENT_NAME = "pdfViewDocumentSaved";

    public PdfViewDocumentSavedEvent(@IdRes int viewId) {
        super(viewId);
    }

    @Override
    public String getEventName() {
        return EVENT_NAME;
    }

    @Override
    public void dispatch(RCTEventEmitter rctEventEmitter) {
        WritableMap eventData = Arguments.createMap();
        rctEventEmitter.receiveEvent(getViewTag(), getEventName(), eventData);
    }
}
